sled-agent/src/instance_manager.rs

```diff
@@ -804,9 +805,13 @@ impl InstanceManagerRunner {
         for (id, instance) in self.jobs.iter() {
             // If we can read the filesystem pool, consider it. Otherwise, move
             // on, to prevent blocking the cleanup of other instances.
-            let Ok(Some(filesystem_pool)) =
-                instance.get_filesystem_zpool().await
-            else {
+            // TODO(eliza): clone each instance and spawn a task to handle it,
+            // so that a single misbehaving instance cannot block the instance
+            // manager's run loop...
+            let (tx, rx) = oneshot::channel();
+            // This will fail if the tx has been dropped, which we just...don't do.
+            let _ = instance.get_filesystem_zpool(tx);
+            let Ok(Ok(Some(filesystem_pool))) = rx.await else {
```

Comment:
Here and in "instance.terminate" below are two spots where we still be seem to block on a per-instance basis.

I don't think that needs to block this change -- it's still a move in the right direction -- but "avoid .await-ing on a single instance" seems like necessary follow-up, otherwise we could still hit the https://github.com/oxidecomputer/omicron/issues/6911 issue, right?

(I see you have the `TODO` here, but just confirming that this also applies to `.terminate`)

Reply:
Yeah, that's correct. I was planning to do that in another branch, but I'm happy to do all of that here, if you prefer?

Reply:
Happy to punt it to another PR, if we're doing it somewhere!